src/utils/index.js

```diff
@@ -0,0 +1,37 @@
+import { WINDOWS_NT, WRITE_TIMEOUT } from "../constants/index.js"
+import { appendFile } from 'fs/promises'
+
+export const getTopProcessCommand = (os) => {
+  switch(os) {
+    case WINDOWS_NT: {
+      return 'powershell "Get-Process | Sort-Object CPU -Descending | Select-Object -Property Name, CPU, WorkingSet -First 1 | ForEach-Object { $_.Name + \' \' + $_.CPU + \' \' + $_.WorkingSet }"'
+    }
+    default: return 'ps -A -o %cpu,%mem,comm | sort -nr | head -n 1'
+  }
+}
+
+export const transformOutput = (output) => {
+  return `${Math.floor(new Date().getTime() / 1000)} : ${output}`
+}
+
+export const getErrorMessage = (error) => {
+  return error instanceof Error ? error.message : String(error)
+}
+
+export const writeToFile = async (filePath, output) => {
+    try {
+      await appendFile(filePath, output)
+    }
+    catch (error) {
+      const errorMessage = getErrorMessage(error)
+      throw Error(`Error writing to file: ${errorMessage}`);
+    }
+}
+
+export const scheduleWriteToFile = (logFilePath, output) => {
+  const timeout = output.text ? WRITE_TIMEOUT : 0
```

Comment:
If I pass value instead of object on calling `scheduleWriteToFile` it will not get it updated in future runs after each minute (the same value will be saved after every minute). My workaround was to pass object into it because I pass the reference so whenever data comes into stream I update the output text and here I always have latest top process running available to add into log file. I was thinking about any much simpler solution but didn't found anything better.